src/mds/Locker.cc

```diff
@@ -2991,6 +2992,10 @@ void Locker::_update_cap_fields(CInode *in, int dirty, MClientCaps *m, inode_t *
 	      << " for " << *in << dendl;
       pi->mtime = mtime;
     }
+    if (((dirty & CEPH_CAP_FILE_WR) && change_attr > pi->change_attr) ||
+	((dirty & CEPH_CAP_FILE_EXCL) && change_attr != pi->change_attr)) {
+      pi->change_attr = change_attr;
```

Comment:
Yeah, I think this is necessary since clients might be changing things under different caps. One might have Fw while another makes a synchronous change under As; we need to bump the change_attr both for the (mds-mediated) inode metadata change and for the data write, but with each bumping it independently from x a ceiling will just keep x+1 despite two separate (and separately-visible) changes.
Or perhaps I'm misunderstanding and simple data writes don't require a bump to the change attr, just changes to file size or something.